pkg/generate/code/set_sdk.go

```diff
@@ -190,6 +190,22 @@ func SetSDK(
 			}
 		}
 
+		fc, ok := fieldConfigs[memberName]
+		if ok && fc.IsSecret {
+			out += fmt.Sprintf("%sif %s.Spec.%s != nil {\n", indent, sourceVarName, memberName)
+			out += fmt.Sprintf("%s%sresp, err := rm.rr.SecretValueFromReference(ctx, %s.Spec.%s)\n", indent,
```

Comment:
Please name the temporary variable containing the secret value something other than `resp`. The name `resp` is already used as the variable name for most HTTP API client call response objects. How about something like `tmpSecret` or something like that?